packages/common/src/content/fetch.ts

```diff
@@ -24,21 +27,42 @@ export interface IFetchContentOptions extends IFetchItemAndEnrichmentsOptions {
   siteOrgKey?: string;
 }
 
+const maybeFetchLayerEnrichments = async (
+  itemAndEnrichments: IItemAndEnrichments,
+  options?: IFetchContentOptions
+) => {
+  // determine if this is a client-side feature layer view
+  const { item, data, layers } = itemAndEnrichments;
+  const layer =
+    layers && getItemLayer(item, layers, options && options.layerId);
+  // TODO: add recordCount here too?
+  const layerEnrichments =
+    layer && isLayerView(layer) && !data
```

Comment:
I copied the current logic in isLayerView() from composer.js, which only checks `layer.isView`, but when creating a new layer view to test this I noticed that the item also gets the "View Service" type keyword. We _could_ use that in `fetchItemEnrichments()` to include "data" in the default list of enrichments to fetch for the item, however, there are layer views that don't have that type keyword (for example the item in [#1073](https://devtopia.esri.com/dc/hub/issues/1073)), so we'd still have to do this check _after_ fetching the layer.